### PR TITLE
Ams sdk

### PR DESCRIPTION
**Pull request by wtrocki**

### CODE REVIEW

File: accountmgmt/go_mod_tidy_hack.go [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package accountmgmt
```
Comment (craicoverflow) [inline accountmgmt/go_mod_tidy_hack.go]: I am not 100% sure, but I think this file was only needed for packages that pre-existed the multi-module system in place. We can leave it in as it won't make a difference, but I do not think that it is needed.

I think you should be able to confirm this by:

1. Remove this file
2. Push to some branch without the file.
3. Run `go get`:

```shell
$ go get github.com/redhat-developer/app-services-sdk-go/accountmgmt@ams-sdk
go: downloading github.com/redhat-developer/app-services-sdk-go/accountmgmt v0.0.0-20220105151805-20fc728e2f7d
go get: added github.com/redhat-developer/app-services-sdk-go/accountmgmt v0.0.0-20220105151805-20fc728e2f7d
```

I also think we can remove it from the other modules also, it was just needed for the first version for each.

File: scripts/generate-static.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+# generate an API client for a service
+generate_sdk() {
+    local file_name=$1
+    local output_path=$2
+    local package_name=$3
+     
+    echo "Validating OpenAPI ${file_name}"
+    npx @openapitools/openapi-generator-cli validate -i "$file_name"
+
```
Comment (craicoverflow) [inline scripts/generate-static.sh]: Why the need for a separate Bash-based generation workflow for this SDK?

File: scripts/generate-static.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    echo "Generating source code based on ${file_name}"
+
+    # remove old generated models
+    rm -Rf $OUTPUT_PATH/model $OUTPUT_PATH/api
+    
+    npx @openapitools/openapi-generator-cli generate -g typescript-axios -i \
```
Comment (craicoverflow) [inline scripts/generate-static.sh]: This is generating TypeScript.

**Review by craicoverflow — COMMENTED** [review]
I can see that the output looks as expected. I left a couple of concerns inline for you to review.

Also - should we split this into a 2-step flow, except we poll the AccountMgmt once per week to check for a new API version? I feel that if we don't have some automated way to check we could fall behind their API.

File: accountmgmt/go_mod_tidy_hack.go (hunk 1 above)
Comment (wtrocki) [reply accountmgmt/go_mod_tidy_hack.go]: Good news. Thank you

Comment (wtrocki) [reply accountmgmt/go_mod_tidy_hack.go]: Will also clean up others after testing it

**Comment by wtrocki** [comment]
> Also - should we split this into a 2-step flow, except we poll the AccountMgmt once per week to check for a new API version? I feel that if we don't have some automated way to check we could fall behind their API.

This api is hosted on gitlab. Any updates will be painful and will need to stay manual due to fact that we use patch. I used patch because we can remove most of their endpoints that we do not use.

AMS will not be documented or updated frequently.
API we are using is stable etc.

File: scripts/generate-static.sh (hunk 3 above)
Comment (wtrocki) [reply scripts/generate-static.sh]: Good catch. Entire method is not used

File: scripts/generate-static.sh (hunk 2 above)
Comment (wtrocki) [reply scripts/generate-static.sh]: Golang one is designed for autoupdates. This is more or less one time script that I will put on package readme for people to discover.

I did tried to add that to golang but that will require pre post hooks defined for each api and some small refactor so gave up

**Comment by craicoverflow** [comment]
> > Also - should we split this into a 2-step flow, except we poll the AccountMgmt once per week to check for a new API version? I feel that if we don't have some automated way to check we could fall behind their API.
> 
> This api is hosted on gitlab

I meant to poll `https://api.openshift.com/api/accounts_mgmt/v1/openapi`.

> Any updates will be painful and will need to stay manual due to fact that we use patch. I used patch because we can remove most of their endpoints that we do not use.
> AMS will not be documented or updated frequently. API we are using is stable etc.

That's fine.

**Review by craicoverflow — APPROVED** [review]

**Pull request merged**